api/v1alpha1/launcherpoolpolicy_types.go

```diff
@@ -0,0 +1,130 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// PoolPolicy defines the proactive provisioning policy for idle launcher pods.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=launcherpoolpolicies,scope=Cluster
+
+type LauncherPoolPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   LauncherPoolPolicySpec   `json:"spec,omitempty"`
+	Status LauncherPoolPolicyStatus `json:"status,omitempty"`
+}
+
+// LauncherPoolPolicySpec defines the node-level idle pool configuration.
+type LauncherPoolPolicySpec struct {
+	// LauncherPoolForNodeType defines pool spec per node type.
+	LauncherPoolForNodeType []LauncherPoolForNodeType `json:"launcherPoolForNodeType"`
+}
+
+// LauncherPoolForNodeType defines launcher count for a class of nodes.
+type LauncherPoolForNodeType struct {
+	// Selector describes the hardware characteristics of target nodes.
+	// +kubebuilder:validation:Required
+	// Introduce an EnhancedNodeSelector that supports combining label-based
+	// matching with resource field conditions.
+	// For example:
+	// enhancedNodeSelector:
+	//  # 1. Label selector (compatible with existing metav1.LabelSelector)
+	//  labelSelector:
+	//    matchLabels:
+	//      nvidia.com/gpu.family: ada-lovelace
+	//    matchExpressions:
+	//      - key: node.kubernetes.io/instance-type
+	//        operator: In
+	//        values: ["gx3-48x240x2l40s", "gx3-96x480x4l40s"]
+	//
+	//  # 2. Resource condition selector (new capability)
+	//  resourceRequirements:
+	//    allocatable:
+	//      memory: ">=200Gi"
+	//      nvidia.com/gpu: ">=1"
+	//      cpu: ">=32"
+	//    # Optional: also support capacity (if total capacity—not just allocatable—is desired)
+	//    # capacity:
+	//    #   memory: ">=256Gi"
+	EnhancedNodeSelector EnhancedNodeSelector `json:"enhancedNodeSelector"`
+
+	// CountForLauncher is the total number of launcher for each LauncherConfig
+	// to maintain on each matching node.
+	// If two different counts are specified for the same (Node, LauncherConfig),
+	// the higher count is used and will be populated into LauncherPoolPolicyStatus.
+	// If no CountForLauncher applies to a given (Node, LauncherConfig), this Node
+	// will be ignored for this LauncherConfig.
+	// +kubebuilder:validation:MinItems=1
+	CountForLauncher []CountForLauncher `json:"countForLauncher"`
+}
+
+// ResourceRequirementSpec defines resource requirements for a node.
+type ResourceRequirementSpec struct {
+	// Allocatable defines the allocatable resources for a node.
+	// +kubebuilder:validation:Required
+	Allocatable map[string]string `json:"allocatable,omitempty"`
```

Comment:
Using one-sided inequalities --- testing only `<=` --- will cause multiple matches where they are not intended. Consider the case of wanting one policy for nodes will a small amount of main memory and a different policy for nodes with a large amount of main memory. If the selector can only test whether the node's main memory is at least as big as a given value, then the policy intended for small nodes will match big ones too.

I recommend instead using a pair of upper and lower bounds. And express the quantities using https://github.com/kubernetes/apimachinery/blob/v0.34.3/pkg/api/resource/quantity.go#L102 , for consistency with how the quantities appear on Node objects.